### PR TITLE
#2559. Add augmenting functions signature tests. Part 2

### PR DESCRIPTION
**Pull request by sgrekhov**

### CODE REVIEW

File: LanguageFeatures/Augmentation-libraries/augmenting_functions_A04_t21.dart [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+  void instanceMethod4({required covariant num i}) {}
+  void instanceMethod5<T>({required covariant T i}) {}
+}
+
+mixin M {
+  void instanceMethod1(num i) {}
+  void instanceMethod2([num i = 1]) {}
+  void instanceMethod3({num i = 1}) {}
+  void instanceMethod4({required num i}) {}
+  void instanceMethod5<T>({required covariant T i}) {}
```
Comment (eernstg) [inline LanguageFeatures/Augmentation-libraries/augmenting_functions_A04_t21.dart:38]: Perhaps?:
```suggestion
  void instanceMethod1(covariant num i) {}
  void instanceMethod2([covariant num i = 1]) {}
  void instanceMethod3({covariant num i = 1}) {}
  void instanceMethod4({required covariant num i}) {}
  void instanceMethod5<T>({required covariant T i}) {}
```

File: LanguageFeatures/Augmentation-libraries/augmenting_functions_A04_t21.dart [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+}
+
+enum E {
+  e1;
+
+  void instanceMethod1(covariant num i) {}
+  void instanceMethod2([num i = 1]) {}
+  void instanceMethod3({num i = 1}) {}
+  void instanceMethod4({required num i}) {}
+  void instanceMethod5<T>({required covariant T i}) {}
```
Comment (eernstg) [inline LanguageFeatures/Augmentation-libraries/augmenting_functions_A04_t21.dart:48]: Perhaps:
```suggestion
  void instanceMethod1(covariant num i) {}
  void instanceMethod2([covariant num i = 1]) {}
  void instanceMethod3({covariant num i = 1}) {}
  void instanceMethod4({required covariant num i}) {}
  void instanceMethod5<T>({required covariant T i}) {}
```

**Review by eernstg — COMMENTED** [review]
Looks good, with one typing issue (we may be unable to maintain that we are getting the error for the correct issue in a few cases), and a couple of inconsistencies (that seem likely to be typos).

**Comment by sgrekhov** [comment]
Updated. Please take another look

**Review by eernstg — APPROVED** [review]
LGTM

**Pull request merged**